### DISCUSSION TITLE
When focus it removes the previously selected value

### DISCUSSION THREAD

**Discussion by bastikun** [Blazored/Typeahead#284] [Ideas]
If possible not to remove the previously selected value
as currently when clicking the textbox or focus on it, it removes the value.

**Comment by chrissainty** [comment]
This happens so you can search for a new value. If you don't select a new value, the old one will show again once you click away.

**Reply by bastikun** [reply]
Yep I mean, once I focus the cursor in textbox the old value should still be there and the X button should appear just like if not in the focus state, the only difference is that the cursor is in focus.
btw thanks for the reply.

**Reply by chrissainty** [reply]
I see what you're saying. Unfortunately the current typeahead design shows a template for the selected item and there isn't an easy way to get at the text that's within that template (RenderFragment). When you click on the typeahead, the selected template is hidden and a input element is rendered.

This can't really be achieved without a rewrite, something that I'd like to do but I just don't have time for at the moment.

**Reply by bastikun** [reply]
Ok thanks for the info. you can close the discussion now. I thought I can achieve also on the JSInterop.
still hoping to fix this in the future when you have time. thanks for this great control.
